pkg/aggregator/aggregator.go

```diff
@@ -315,6 +326,27 @@ func (agg *BufferedAggregator) flushSeries() {
 	start := time.Now()
 	series := agg.GetSeries()
 
+	extraSeriesLock.Lock()
+	// Adding extraSeries to the flushed ones
+	for _, extra := range extraSeries {
+		if extra.Host == "" {
+			extra.Host = agg.hostname
+		}
+		if extra.SourceTypeName == "" {
+			extra.SourceTypeName = "System"
+		}
+
+		// Updateting Ts for every points
+		updatedPoints := []metrics.Point{}
+		for _, point := range extra.Points {
+			point.Ts = float64(start.Unix())
+			updatedPoints = append(updatedPoints, point)
+		}
+		extra.Points = updatedPoints
+		series = append(series, extra)
+	}
+	extraSeriesLock.Unlock()
+
 	// Send along a metric that showcases that this Agent is running (internally, in backend,
 	// a `datadog.`-prefixed metric allows identifying this host as an Agent host, used for dogbone icon)
 	series = append(series, &metrics.Serie{
```

Comment:
would it make sense to add the 2 metrics defined below to `extraSeries`?

Reply:
I thought about it.

We can't for `n_o_i_n_d_e_x.datadog.%s.payload.dropped` because the value change at every flushed and since this is kind of a workaround while we design a real telemetry system I don't want to make something over-complex with callback for the metric value.

For `datadog.%s.running` we could but it doesn't bring any value I think.

Wdyt ?

Reply:
thanks for the explanation for the `n_o_i_n_d_e_x.datadog.%s.payload.dropped` 👍  Could you add a comment on that code to explain that? I agree that complexifying the design only to allow supporting this metric isn't worth it.

For `datadog.%s.running` I'd say it's worth doing to keep things consistent (I think it'd make the code easier to read and understand)

Reply:
Just realized that `datadog.%s.running` need the `agentName` that is provided when we initialize the aggregator.

`recurrentSeries` are package variables set from other packages. This means that if we create multiple aggregator, `recurrentSeries` are shared by all aggregator instances. If we add `datadog.%s.running` to it, all aggregator instances would send the metric with the first or latest instantiated `agentName` (depending on the implementation). Even if multiple aggregators is unlikely it make more sens if each send that metrics with the correct tag.

Reply:
I see, then would it make sense to make `recurrentSeries` an instance-level variable of an aggregator? With maybe a package-level exported function that allows adding recurrent series to the current aggregator singleton.

Would likely also make testing easier (would love to see tests on this btw :) )

Reply:
We have to think about init order here. The aggregator might not be initialized when we add the first add a recurrent serie. That's why is a package level variable and not a instance level one. It also play along having an API that allows multiple aggregator.

I'll add some test !